sympy/physics/control/lti.py

```diff
@@ -3838,6 +3840,108 @@ def num_outputs(self):
         """
         return self._D.rows
 
+    def state_vector(self, initial_conditions=None, input_vector=None, var=None):
+        r"""
+        Returns the State vector `x` given by the solution of the state equation
+            x'(t) = A * x(t) + B * u(t)
+
+        Parameters
+        ============
+
+        initial_conditions : Matrix
+            The initial conditions of `x` state vector.
+        input_vector : Matrix
+            The input vector for state space.
+        var : Symbol
+            The symbol representing time.
+
+        Examples
+        ==========
+
+        >>> from sympy import Matrix, Symbol
+        >>> from sympy.physics.control import StateSpace
+        >>> A = Matrix([[-2, 0], [1, -1]])
+        >>> i = Matrix([2, 3])
+        >>> t = Symbol('t')
+        >>> ss = StateSpace(A)
+        >>> ss.state_vector(initial_conditions=i, var=t)
+        Matrix([
+        [            2*exp(-2*t)],
+        [5*exp(-t) - 2*exp(-2*t)]])
+
+        References
+        ==========
+        .. [1] https://web.mit.edu/2.14/www/Handouts/StateSpaceResponse.pdf
+        .. [2] https://docs.sympy.org/latest/modules/solvers/ode.html#sympy.solvers.ode.systems.linodesolve
+        """
+
+        if not var:
+            var = Symbol('t')
+        elif not isinstance(var, Symbol):
+            raise ValueError("Variable for representing time must be a Symbol.")
+        if not initial_conditions:
+            initial_conditions = zeros(self._A.shape[0], 1)
+        elif initial_conditions.shape != (self._A.shape[0], 1):
+            raise ShapeError("Initial condition vector should have the same number of "
+                             "rows as the state matrix.")
+        if not input_vector:
+            input_vector = zeros(self._B.shape[1], 1)
+        elif input_vector.shape != (self._B.shape[1], 1):
+            raise ShapeError("Input vector should have the same number of "
+                             "columns as the input matrix.")
+        sol = linodesolve(self._A, var, self._B*input_vector, type='type2', doit=True)
+        mat1 = Matrix(sol)
+        mat2 = mat1.replace(var, 0)
+        # Get all the free symbols form the matrix
+        dummy_symbols = list(mat2.free_symbols)
+        # Convert the matrix to a Coefficient matrix
+        r1, r2 = linear_eq_to_matrix(mat2, dummy_symbols)
+        s = linsolve((r1, initial_conditions+r2))
+        res_tuple = next(iter(s))
+        for ind, v in enumerate(res_tuple):
+            mat1 = mat1.replace(dummy_symbols[ind], v)
+        return mat1
+
+    def output_vector(self, initial_conditions=None, input_vector=None, var=None):
```

Comment:
I'm not a huge fan of the names of these new methods. You are solving the linear ordinary differential equations initial value problem. `initial_value_problem_solution` or `ivp_sol` or `solve_ivp` (which is same as scipy's numerical method seem more aligned with what the methods actually do.

Reply:
+1

Reply:
We also already have `dsolve()` terminology in SymPy, so copying what we already have for these names is also probably a good idea.

Reply:
Do you mean that the `output_vector()` should be renamed as `dsolve()` or `ivp_sol()` ?

Reply:
Yes, both `state_vector` and `output_vector` coudl have better names. You also do not need a `state_vector` method because you can always pass `C = eye(len(A))` to make `y = x`. I would only expose the single method to the users and name it something that aligns with the naming already in sympy or the name in scipy.